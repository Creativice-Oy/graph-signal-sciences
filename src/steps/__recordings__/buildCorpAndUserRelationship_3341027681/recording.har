{
  "log": {
    "_recordingName": "buildCorpAndUserRelationship",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "a0d858918dc7000c5fe61c66948ecc2e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "x-api-user",
              "value": "[REDACTED]"
            },
            {
              "name": "x-api-token",
              "value": "[REDACTED]"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "user-agent",
              "value": "axios/0.25.0"
            },
            {
              "name": "host",
              "value": "dashboard.signalsciences.net"
            }
          ],
          "headersSize": 297,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://dashboard.signalsciences.net/api/v0/corps"
        },
        "response": {
          "bodySize": 435,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 435,
            "text": "{\"data\":[{\"name\":\"jupiterone\",\"displayName\":\"JupiterOne\",\"smallIconURI\":\"\",\"created\":\"2022-01-10T18:29:55Z\",\"siteLimit\":5,\"sites\":{\"uri\":\"/api/v0/corps/jupiterone/sites\"},\"authType\":\"builtin\",\"logoutURI\":\"\",\"samlCert\":\"\",\"samlEndpoint\":\"\",\"signRequestsUsingStoredCert\":false,\"samlRequestCert\":\"\",\"sessionMaxAgeDashboard\":2592000,\"apiAllowedIPs\":[],\"apiTokenMaxAge\":0,\"restrictedAccessTokens\":false,\"ssoProvisioningConfigured\":false}]}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Mon, 24 Jan 2022 21:05:37 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "435"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, no-cache, no-store"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-request-id",
              "value": "61ef14a157a6a801e6f02233"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block; report=/csp/reports"
            }
          ],
          "headersSize": 395,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-01-24T21:05:37.423Z",
        "time": 309,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 309
        }
      },
      {
        "_id": "a1ad97198eead933a970472a10fd77f9",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "x-api-user",
              "value": "[REDACTED]"
            },
            {
              "name": "x-api-token",
              "value": "[REDACTED]"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "user-agent",
              "value": "axios/0.25.0"
            },
            {
              "name": "host",
              "value": "dashboard.signalsciences.net"
            }
          ],
          "headersSize": 314,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://dashboard.signalsciences.net/api/v0/corps/jupiterone/users"
        },
        "response": {
          "bodySize": 2181,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 2181,
            "text": "{\"data\":[{\"name\":\"Samuel Poulton Observer\",\"email\":\"samuel.poulton+sigsci_observer@jupiterone.com\",\"announcements\":{\"uri\":\"/api/v0/user/announcements\"},\"defaultDashboards\":{\"uri\":\"/api/v0/user/defaultDashboards\"},\"memberships\":{\"data\":[],\"uri\":\"/api/v0/corps/jupiterone/users/samuel.poulton+sigsci_observer@jupiterone.com/memberships\"},\"accessTokens\":{\"uri\":\"/api/v0/user/apitokens\"},\"role\":\"observer\",\"status\":\"active\",\"mfaEnabled\":false,\"corpAuthType\":\"builtin\",\"authStatus\":\"none\",\"created\":\"2022-01-19T21:49:48.458Z\",\"apiUser\":false,\"accessTokenAllowed\":true,\"ssoProvisioned\":false},{\"name\":\"Samuel Poulton\",\"email\":\"samuel.poulton@jupiterone.com\",\"announcements\":{\"uri\":\"/api/v0/user/announcements\"},\"defaultDashboards\":{\"uri\":\"/api/v0/user/defaultDashboards\"},\"memberships\":{\"data\":[],\"uri\":\"/api/v0/corps/jupiterone/users/samuel.poulton@jupiterone.com/memberships\"},\"accessTokens\":{\"uri\":\"/api/v0/user/apitokens\"},\"role\":\"owner\",\"status\":\"active\",\"mfaEnabled\":false,\"corpAuthType\":\"builtin\",\"authStatus\":\"none\",\"created\":\"2022-01-18T17:51:41.435Z\",\"apiUser\":false,\"accessTokenAllowed\":true,\"ssoProvisioned\":false},{\"name\":\"Sam Andrus\",\"email\":\"sam.andrus@jupiterone.com\",\"announcements\":{\"uri\":\"/api/v0/user/announcements\"},\"defaultDashboards\":{\"uri\":\"/api/v0/user/defaultDashboards\"},\"memberships\":{\"data\":[],\"uri\":\"/api/v0/corps/jupiterone/users/sam.andrus@jupiterone.com/memberships\"},\"accessTokens\":{\"uri\":\"/api/v0/user/apitokens\"},\"role\":\"owner\",\"status\":\"active\",\"mfaEnabled\":false,\"corpAuthType\":\"builtin\",\"authStatus\":\"none\",\"created\":\"2022-01-10T18:30:40.62Z\",\"apiUser\":false,\"accessTokenAllowed\":true,\"ssoProvisioned\":false},{\"name\":\"Austin Kelleher\",\"email\":\"austin.kelleher@jupiterone.com\",\"announcements\":{\"uri\":\"/api/v0/user/announcements\"},\"defaultDashboards\":{\"uri\":\"/api/v0/user/defaultDashboards\"},\"memberships\":{\"data\":[],\"uri\":\"/api/v0/corps/jupiterone/users/austin.kelleher@jupiterone.com/memberships\"},\"accessTokens\":{\"uri\":\"/api/v0/user/apitokens\"},\"role\":\"owner\",\"status\":\"active\",\"mfaEnabled\":false,\"corpAuthType\":\"builtin\",\"authStatus\":\"none\",\"created\":\"2022-01-10T18:30:28.296Z\",\"apiUser\":false,\"accessTokenAllowed\":true,\"ssoProvisioned\":false}]}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Mon, 24 Jan 2022 21:05:38 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "transfer-encoding",
              "value": "chunked"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, no-cache, no-store"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-request-id",
              "value": "61ef14a229af8201e7c417e9"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block; report=/csp/reports"
            }
          ],
          "headersSize": 402,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-01-24T21:05:37.764Z",
        "time": 174,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 174
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
