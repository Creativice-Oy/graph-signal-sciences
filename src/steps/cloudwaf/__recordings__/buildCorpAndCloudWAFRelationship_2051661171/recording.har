{
  "log": {
    "_recordingName": "buildCorpAndCloudWAFRelationship",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "79c71d206a4c9cf236c090578b230d37",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 64,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "name": "user-agent",
              "value": "axios/0.25.0"
            },
            {
              "name": "content-length",
              "value": 64
            },
            {
              "name": "host",
              "value": "dashboard.signalsciences.net"
            }
          ],
          "headersSize": 240,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "email=sam.andrus%40jupiterone.com&password=DXW4mwz%40vab*gkd1mqf"
          },
          "queryString": [],
          "url": "https://dashboard.signalsciences.net/api/v0/auth"
        },
        "response": {
          "bodySize": 49,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 49,
            "text": "{\"token\":\"475f59f2-966e-4bfe-bb5e-713854750fda\"}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Mon, 07 Nov 2022 13:51:56 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "49"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, no-cache, no-store"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-request-id",
              "value": "63690d7cdafb4ecfcbae4910"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block; report=/csp/reports"
            }
          ],
          "headersSize": 394,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-11-07T13:51:56.019Z",
        "time": 834,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 834
        }
      },
      {
        "_id": "556a3fcc8d26abfc35cb34522cd84237",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "name": "user-agent",
              "value": "axios/0.25.0"
            },
            {
              "name": "host",
              "value": "dashboard.signalsciences.net"
            }
          ],
          "headersSize": 260,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://dashboard.signalsciences.net/api/v0/corps/jupiterone/cloudwafInstances"
        },
        "response": {
          "bodySize": 1661,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 1661,
            "text": "{\"data\":[{\"id\":\"9adf79f2-6553-4008-a012-e2e78d1c40eb\",\"name\":\"testinstance\",\"description\":\"testinstance\",\"region\":\"us-east-1\",\"tlsMinVersion\":\"1.2\",\"workspaceConfigs\":[{\"siteName\":\"test\",\"instanceLocation\":\"direct\",\"clientIPHeader\":\"\",\"listenerProtocols\":[\"https\"],\"routes\":[{\"certificateIds\":[],\"domains\":[\"paradox-test.blogspot.com\"],\"origin\":\"https://paradox-test.blogspot.com/\",\"passHostHeader\":false,\"id\":\"4d14d79e-1f1b-4c8f-a662-1c07ff4c46db\",\"connectionPooling\":true,\"trustProxyHeaders\":false}]}],\"deployment\":{\"status\":\"done\",\"message\":\"\",\"egressIPs\":[{\"ip\":\"54.157.9.221\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.211.133.174\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"44.198.49.148\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.222.240.59\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"44.198.76.176\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"34.192.82.71\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.216.58.131\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"18.214.162.196\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"54.84.118.7\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"52.204.194.140\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.232.125.243\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"107.22.39.13\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"54.147.50.184\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"54.160.112.153\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.227.47.195\",\"status\":\"unknown\",\"updatedAt\":\"\"},{\"ip\":\"3.208.110.204\",\"status\":\"unknown\",\"updatedAt\":\"\"}],\"dnsEntry\":\"9adf79f2-6553-4008-a012-e2e78d1c40eb.sigscicloudwaf.com\"},\"useUploadedCertificates\":false,\"createdBy\":\"sam.andrus@jupiterone.com\",\"created\":\"2022-11-07T05:27:11Z\",\"updatedBy\":\"\"}]}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Mon, 07 Nov 2022 13:51:57 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "1661"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, no-cache, no-store"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-request-id",
              "value": "63690d7d44e822837ef399e3"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block; report=/csp/reports"
            }
          ],
          "headersSize": 396,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-11-07T13:51:56.858Z",
        "time": 802,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 802
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
