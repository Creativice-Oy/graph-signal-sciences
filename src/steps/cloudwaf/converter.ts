import {
  createIntegrationEntity,
  createDirectRelationship,
  parseTimePropertyValue,
  Entity,
  RelationshipClass,
  Relationship,
} from '@jupiterone/integration-sdk-core';

import { Entities } from '../constants';
import { SignalSciencesCloudWAF } from '../../types';

export function getCloudWAFKey(id: string): string {
  return `sigsci_cloudwaf:${id}`;
}

export function createCloudWAFEntity(cloudWAF: SignalSciencesCloudWAF): Entity {
  return createIntegrationEntity({
    entityData: {
      source: cloudWAF,
      assign: {
        _type: Entities.CLOUD_WAF._type,
        _class: Entities.CLOUD_WAF._class,
        _key: getCloudWAFKey(cloudWAF.id),
        name: cloudWAF.name,
        displayName: cloudWAF.name,
        category: ['application'],
        description: cloudWAF.description,
        region: cloudWAF.region,
        tlsMinVersion: cloudWAF.tlsMinVersion,
        siteName: cloudWAF.workspaceConfigs[0].siteName,
        'deployment.Status': cloudWAF.deployment.status,
        'deployment.Message': cloudWAF.deployment.message,
        'deployment.EgressIps': cloudWAF.deployment.egressIPs.map(
          (egressIp) => egressIp.ip,
        ),
        'deployment.DnsEntry': cloudWAF.deployment.dnsEntry,
        useUploadedCertificates: cloudWAF.useUploadedCertificates,
        createdBy: cloudWAF.createdBy,
        createdOn: parseTimePropertyValue(cloudWAF.created),
        updatedBy: cloudWAF.updatedBy,
      },
    },
  });
}

export function createOrganizationCloudwafRelationship(
  orgainzation: Entity,
  cloudWaf: Entity,
): Relationship {
  return createDirectRelationship({
    _class: RelationshipClass.HAS,
    from: orgainzation,
    to: cloudWaf,
  });
}
