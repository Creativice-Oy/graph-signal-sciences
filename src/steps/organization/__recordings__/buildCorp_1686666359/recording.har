{
  "log": {
    "_recordingName": "buildCorp",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "5.1.1"
    },
    "entries": [
      {
        "_id": "a0d858918dc7000c5fe61c66948ecc2e",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "x-api-user",
              "value": "[REDACTED]"
            },
            {
              "name": "x-api-token",
              "value": "[REDACTED]"
            },
            {
              "name": "content-type",
              "value": "application/json"
            },
            {
              "name": "user-agent",
              "value": "axios/0.25.0"
            },
            {
              "name": "host",
              "value": "dashboard.signalsciences.net"
            }
          ],
          "headersSize": 297,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://dashboard.signalsciences.net/api/v0/corps"
        },
        "response": {
          "bodySize": 435,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 435,
            "text": "{\"data\":[{\"name\":\"jupiterone\",\"displayName\":\"JupiterOne\",\"smallIconURI\":\"\",\"created\":\"2022-01-10T18:29:55Z\",\"siteLimit\":5,\"sites\":{\"uri\":\"/api/v0/corps/jupiterone/sites\"},\"authType\":\"builtin\",\"logoutURI\":\"\",\"samlCert\":\"\",\"samlEndpoint\":\"\",\"signRequestsUsingStoredCert\":false,\"samlRequestCert\":\"\",\"sessionMaxAgeDashboard\":2592000,\"apiAllowedIPs\":[],\"apiTokenMaxAge\":0,\"restrictedAccessTokens\":false,\"ssoProvisioningConfigured\":false}]}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Mon, 24 Jan 2022 23:27:22 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "435"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, no-cache, no-store"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-request-id",
              "value": "61ef35da2505ff01e69ae136"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block; report=/csp/reports"
            }
          ],
          "headersSize": 395,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-01-24T23:27:21.832Z",
        "time": 301,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 301
        }
      },
      {
        "_id": "79c71d206a4c9cf236c090578b230d37",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 64,
          "cookies": [],
          "headers": [
            {
              "name": "accept",
              "value": "application/json, text/plain, */*"
            },
            {
              "name": "content-type",
              "value": "application/x-www-form-urlencoded"
            },
            {
              "name": "user-agent",
              "value": "axios/0.25.0"
            },
            {
              "name": "content-length",
              "value": 64
            },
            {
              "name": "host",
              "value": "dashboard.signalsciences.net"
            }
          ],
          "headersSize": 240,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/x-www-form-urlencoded",
            "params": [],
            "text": "email=sam.andrus%40jupiterone.com&password=DXW4mwz%40vab*gkd1mqf"
          },
          "queryString": [],
          "url": "https://dashboard.signalsciences.net/api/v0/auth"
        },
        "response": {
          "bodySize": 49,
          "content": {
            "mimeType": "application/json; charset=utf-8",
            "size": 49,
            "text": "{\"token\":\"2d87ef01-ef99-44c9-af0f-04a58f7b00d4\"}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "date",
              "value": "Mon, 07 Nov 2022 13:51:56 GMT"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=utf-8"
            },
            {
              "name": "content-length",
              "value": "49"
            },
            {
              "name": "connection",
              "value": "close"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, no-cache, no-store"
            },
            {
              "name": "strict-transport-security",
              "value": "max-age=31536000; includeSubDomains"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "x-frame-options",
              "value": "deny"
            },
            {
              "name": "x-request-id",
              "value": "63690d7c78a2902aa896adb4"
            },
            {
              "name": "x-xss-protection",
              "value": "1; mode=block; report=/csp/reports"
            }
          ],
          "headersSize": 394,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2022-11-07T13:51:55.932Z",
        "time": 896,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 896
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
