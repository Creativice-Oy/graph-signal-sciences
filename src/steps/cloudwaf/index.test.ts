import {
  createMockStepExecutionContext,
  Recording,
} from '@jupiterone/integration-sdk-testing';

import { setupSigSciRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../config';
import { fetchCloudWAF } from './';
import { integrationConfig } from '../../../test/config';
import { createOrganizationEntity } from '../organization/converter';
import { SignalSciencesCorp } from '../../types';

describe('should collect cloudwaf instance data', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupSigSciRecording({
      directory: __dirname,
      name: 'buildCorpAndCloudWAFRelationship',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('validate client integration and converters', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });
    await context.jobState.addEntity(
      createOrganizationEntity({
        name: 'jupiterone',
        displayName: 'JupiterOne',
      } as SignalSciencesCorp),
    );

    await fetchCloudWAF(context);

    expect({
      numCollectedEntities: context.jobState.collectedEntities.length,
      numCollectedRelationships: context.jobState.collectedRelationships.length,
      collectedEntities: context.jobState.collectedEntities,
      collectedRelationships: context.jobState.collectedRelationships,
      encounteredTypes: context.jobState.encounteredTypes,
    }).toMatchSnapshot();

    const cloudwafInstances = context.jobState.collectedEntities.filter((e) =>
      e._class.includes('Firewall'),
    );
    expect(cloudwafInstances.length).toBeGreaterThan(0);
    expect(cloudwafInstances).toMatchGraphObjectSchema({
      _class: ['Firewall'],
      schema: {
        additionalProperties: false,
        properties: {
          _type: { const: 'sigsci_cloudwaf' },
          _rawData: {
            type: 'array',
            items: { type: 'object' },
          },
          name: { type: 'string' },
          displayName: { type: 'string' },
          category: {
            type: 'array',
            items: { type: 'string' },
          },
          description: { type: 'string' },
          region: { type: 'string' },
          tlsMinVersion: { type: 'string' },
          siteName: { type: 'string' },
          'deployment.Status': { type: 'string' },
          'deployment.Message': { type: 'string' },
          'deployment.EgressIps': {
            type: 'array',
            items: { type: 'string' },
          },
          'deployment.DnsEntry': { type: 'string' },
          useUploadedCertificates: { type: 'boolean' },
          createdBy: { type: 'string' },
          createdOn: { type: 'number' },
          updatedBy: { type: 'string' },
        },
      },
    });
  });
});
