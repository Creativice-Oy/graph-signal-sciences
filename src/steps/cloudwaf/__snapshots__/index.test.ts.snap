// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should collect cloudwaf instance data validate client integration and converters 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "Organization",
      ],
      "_key": "jupiterone",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "displayName": "JupiterOne",
            "name": "jupiterone",
          },
        },
      ],
      "_type": "sigsci_corp",
      "apiTokenMaxAge": undefined,
      "authType": undefined,
      "createdOn": undefined,
      "displayName": "JupiterOne",
      "logoutURI": undefined,
      "name": "jupiterone",
      "restrictedAccessTokens": undefined,
      "samlCert": undefined,
      "samlRequestCert": undefined,
      "sessionMaxAgeDashboard": undefined,
      "signRequestsUsingStoredCert": undefined,
      "siteLimit": undefined,
      "sitesUri": undefined,
      "smallIconURI": undefined,
      "ssoProvisioningConfigured": undefined,
      "webLink": "https://dashboard.signalsciences.net/corps/jupiterone/overview",
    },
    Object {
      "_class": Array [
        "Firewall",
      ],
      "_key": "sigsci_cloudwaf:9adf79f2-6553-4008-a012-e2e78d1c40eb",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "created": "2022-11-07T05:27:11Z",
            "createdBy": "sam.andrus@jupiterone.com",
            "deployment": Object {
              "dnsEntry": "9adf79f2-6553-4008-a012-e2e78d1c40eb.sigscicloudwaf.com",
              "egressIPs": Array [
                Object {
                  "ip": "54.157.9.221",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "3.211.133.174",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "44.198.49.148",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "3.222.240.59",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "44.198.76.176",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "34.192.82.71",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "3.216.58.131",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "18.214.162.196",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "54.84.118.7",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "52.204.194.140",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "3.232.125.243",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "107.22.39.13",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "54.147.50.184",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "54.160.112.153",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "3.227.47.195",
                  "status": "unknown",
                  "updatedAt": "",
                },
                Object {
                  "ip": "3.208.110.204",
                  "status": "unknown",
                  "updatedAt": "",
                },
              ],
              "message": "",
              "status": "done",
            },
            "description": "testinstance",
            "id": "9adf79f2-6553-4008-a012-e2e78d1c40eb",
            "name": "testinstance",
            "region": "us-east-1",
            "tlsMinVersion": "1.2",
            "updatedBy": "",
            "useUploadedCertificates": false,
            "workspaceConfigs": Array [
              Object {
                "clientIPHeader": "",
                "instanceLocation": "direct",
                "listenerProtocols": Array [
                  "https",
                ],
                "routes": Array [
                  Object {
                    "certificateIds": Array [],
                    "connectionPooling": true,
                    "domains": Array [
                      "paradox-test.blogspot.com",
                    ],
                    "id": "4d14d79e-1f1b-4c8f-a662-1c07ff4c46db",
                    "origin": "https://paradox-test.blogspot.com/",
                    "passHostHeader": false,
                    "trustProxyHeaders": false,
                  },
                ],
                "siteName": "test",
              },
            ],
          },
        },
      ],
      "_type": "sigsci_cloudwaf",
      "category": Array [
        "application",
      ],
      "createdBy": "sam.andrus@jupiterone.com",
      "createdOn": 1667798831000,
      "deployment.DnsEntry": "9adf79f2-6553-4008-a012-e2e78d1c40eb.sigscicloudwaf.com",
      "deployment.EgressIps": Array [
        "54.157.9.221",
        "3.211.133.174",
        "44.198.49.148",
        "3.222.240.59",
        "44.198.76.176",
        "34.192.82.71",
        "3.216.58.131",
        "18.214.162.196",
        "54.84.118.7",
        "52.204.194.140",
        "3.232.125.243",
        "107.22.39.13",
        "54.147.50.184",
        "54.160.112.153",
        "3.227.47.195",
        "3.208.110.204",
      ],
      "deployment.Message": "",
      "deployment.Status": "done",
      "description": "testinstance",
      "displayName": "testinstance",
      "id": "9adf79f2-6553-4008-a012-e2e78d1c40eb",
      "name": "testinstance",
      "region": "us-east-1",
      "siteNames": Array [
        "test",
      ],
      "tlsMinVersion": "1.2",
      "updatedBy": "",
      "useUploadedCertificates": false,
    },
  ],
  "collectedRelationships": Array [
    Object {
      "_class": "HAS",
      "_fromEntityKey": "jupiterone",
      "_key": "jupiterone|has|sigsci_cloudwaf:9adf79f2-6553-4008-a012-e2e78d1c40eb",
      "_toEntityKey": "sigsci_cloudwaf:9adf79f2-6553-4008-a012-e2e78d1c40eb",
      "_type": "sigsci_corp_has_cloudwaf",
      "displayName": "HAS",
    },
  ],
  "encounteredTypes": Array [
    "sigsci_corp",
    "sigsci_cloudwaf",
    "sigsci_corp_has_cloudwaf",
  ],
  "numCollectedEntities": 2,
  "numCollectedRelationships": 1,
}
`;
